Mathlib/CategoryTheory/MorphismProperty/Lifting.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2024 Jack McKoen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jack McKoen
+-/
+import Mathlib.CategoryTheory.MorphismProperty.Limits
+import Mathlib.CategoryTheory.MorphismProperty.Retract
+
+/-!
+# Left and right lifting properties
+
+Given a morphism property `T`, we define the left and right lifting property with respect to `T`.
+
+We show that the left lifting property is stable under retracts, cobase change, and composition,
+with dual statements for the right lifting property.
+
+-/
+
+universe v u
+
+namespace CategoryTheory
+
+variable {C : Type u} [Category.{v} C]
+
+namespace MorphismProperty
+
+/-- The left lifting property (llp) with respect to a class of morphisms. -/
+def llp (T : MorphismProperty C) : MorphismProperty C := fun _ _ f ↦
+  ∀ ⦃X Y : C⦄ ⦃g : X ⟶ Y⦄ (_ : T g), HasLiftingProperty f g
```

Comment:
In the applications, I believe it will be easier to have `g` as an explicit parameter.
```suggestion
  ∀ ⦃X Y : C⦄ (g : X ⟶ Y) (_ : T g), HasLiftingProperty f g
```
(The idea is that when we want to apply `llp`, we feed `g`, which creates the goal `T g`.)